### ISSUE TITLE
Scale (and Mode) Identification Exercise

### ISSUE THREAD

**Issue by RustoMCSpit** [ShacharHarshuv/open-ear#234]
a random midi section will play and your job is to identiy the scale and mode played

**Comment by ShacharHarshuv** [comment]
We need to define what actually should be played. Is it just the scale going up? If it's a more arbitrary melody, we need to generate one so that the information is clear. The mode cannot be determined from the notes alone.

**Comment by RustoMCSpit** [comment]
> We need to define what actually should be played. Is it just the scale going up? If it's a more arbitrary melody, we need to generate one so that the information is clear. The mode cannot be determined from the notes alone, even if you play all notes.

i believe the more settings available to the user the better, the melody can be coded in a way that there are cadences in it. i think the best approach is to download a huge midi pack file and transpose to the desired mode.

**Comment by bokajsukriz** [comment]
+1
Also one idea: if there would be options to choose (the standard progression (I-IV-V-I), a random melody, a random progression, random progression + melody) the same options could maybe be adopted to other exercices. So instead of the standard progression you could choose to get your ears set for the new scale by a random progression / melody?

**Comment by ShacharHarshuv** [comment]
I think the easiest thing to do is to compose a small set of cadential melodies with a clear scale definition. However - will learning these necessarily help the user learn to identify scales in real music?

**Comment by ShacharHarshuv** [comment]
Correct, it is not possible to identify a scale by random notes, as I explain above. 
However it might be possible to compose a set of cadential melodies in every scale that will be enough information to identify a scale.

Also - perfect pitch is not needed if you're not interested in the absolute root note. I thought the meaning was "identify major scale vs minor scale" rather than "identify C major vs D major".

Tbh, I feel like you're better off just practicing melody recognition.

**Comment by ShacharHarshuv** [comment]
Closing as I see little musical pedagogical value in this proposed exercise, and it's not well defined. 
That being said, if you really want to implement this you can try to open a PR and I'm open to merging it.
